akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolConductor.scala

```diff
@@ -140,15 +140,15 @@ private object PoolConductor {
               slotStates(slotIx) = slotStateAfterDisconnect(slotStates(slotIx), failed)
           }
           pull(slotIn)
-          val tryPull = nextSlot == -1
+          val wasBlocked = nextSlot == -1
           nextSlot = bestSlot()
-          if (tryPull) tryPullCtx()
+          if (wasBlocked && nextSlot != -1) pull(ctxIn)
```

Comment:
ah right, this is the slot port. Maybe name `nextSlot != -1` to `nowUnblocked` and add a comment:

    if (wasBlocked && nowUnblocked) pull(ctxIn) // Get next request context

Not to nitpick, but this is mostly alien code to me so any help is nice to have :)